File: android/java/AppsPrizeDemo/app/src/main/java/com/appsamurai/appsprizedemo/MainActivity.java
package com.appsamurai.appsprizedemo;

import android.app.Activity;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;

import androidx.annotation.Nullable;

import com.appsamurai.appsprize.AppReward;
import com.appsamurai.appsprize.AppsPrize;
import com.appsamurai.appsprize.AppsPrizeRewardListener;

import java.util.List;

public class MainActivity extends Activity {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Button button = findViewById(R.id.offers_button);

        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                boolean available = AppsPrize.launchActivity(MainActivity.this);
                if (!available) {
                    Log.e("[AppsPrize]", "AppsPrize not initialized");
                }
            }
        });

        AppsPrize.doReward(MainActivity.this, new AppsPrizeRewardListener() {
            @Override
            public void onSessionReward(List<AppReward> list) {
                // used for handling session reward without AppsPrizeListener.onRewardUpdate event method
            }
        });
    }
}

File: android/kotlin/AppsPrizeDemo/app/src/main/java/com/appsamurai/appsprize_demo/MainActivity.kt
package com.appsamurai.appsprize_demo

import android.app.Activity
import android.os.Bundle
import android.util.Log
import com.appsamurai.appsprize.AppsPrize
import com.appsamurai.appsprize_demo.databinding.ActivityMainBinding

class MainActivity : Activity() {

    private lateinit var binding: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        binding.offersButton.setOnClickListener {
            val available = AppsPrize.launchActivity(this@MainActivity)
            if (!available) {
                Log.e("[AppsPrize]", "AppsPrize not initialized")
            }
        }

        AppsPrize.doReward(this) { sessionReward ->
            // used for handling session reward without AppsPrizeListener.onRewardUpdate event method
        }
    }
}